pkg/sql/executor.go

```diff
@@ -1078,6 +1091,25 @@ func (e *Executor) noTransactionHelper(txnState *txnState) (Result, error) {
 	return Result{Err: errNoTransactionInProgress}, errNoTransactionInProgress
 }
 
+// showTransactionState returns the state of current transaction.
+func showTransactionState(txnState *txnState, implicitTxn bool, planMaker *planner) (Result, error) {
```

Comment:
1) rename to `runShowTransactionState`
2) make this function a method of planner: `func (p *planner) runShowTransactionState(...)`